File: DSA/Arrays/Code/7_ArrayUpdate.java
public class ArrayUpdate {
    // Array Update in Java
    // Modify/Update elements at specific positions
    
    static void updateElement(int[] arr, int position, int newValue) {
        arr[position] = newValue;
    }
    
    static void displayArray(int[] arr) {
        for (int val : arr) {
            System.out.print(val + " ");
        }
        System.out.println();
    }
    
    public static void main(String[] args) {
        int[] arr = {10, 20, 30, 40, 50};
        
        System.out.print("Original array: ");
        displayArray(arr);
        
        // Method 1: Direct update
        arr[0] = 15;
        System.out.print("After updating index 0 to 15: ");
        displayArray(arr);
        
        // Method 2: Using function
        updateElement(arr, 2, 35);
        System.out.print("After updating index 2 to 35: ");
        displayArray(arr);
        
        // Method 3: Conditional update
        arr = new int[]{10, 20, 30, 40, 50};
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] > 20) {
                arr[i] *= 2;
            }
        }
        System.out.print("After doubling elements > 20: ");
        displayArray(arr);
        
        // Method 4: Update all elements
        arr = new int[]{10, 20, 30, 40, 50};
        int addValue = 5;
        for (int i = 0; i < arr.length; i++) {
            arr[i] += addValue;
        }
        System.out.print("After adding 5 to all elements: ");
        displayArray(arr);
        
        // Method 5: Update using stream (Java 8+)
        arr = new int[]{10, 20, 30, 40, 50};
        for (int i = 0; i < arr.length; i++) {
            arr[i] = arr[i] * 2;
        }
        System.out.print("After doubling all elements: ");
        displayArray(arr);
    }
}

File: DSA/Arrays/Code/7_ArrayUpdate.py
# Array Update in Python
# Modify/Update elements at specific positions

arr = [10, 20, 30, 40, 50]

print("Original array:", arr)

# Method 1: Direct update by index
arr[0] = 15
print("After updating index 0 to 15:", arr)

# Method 2: Update multiple elements
arr[2] = 35
arr[4] = 55
print("After updating indices 2 and 4:", arr)

# Method 3: Conditional update
arr = [10, 20, 30, 40, 50]
for i in range(len(arr)):
    if arr[i] > 20:
        arr[i] *= 2
print("After doubling elements > 20:", arr)

# Method 4: Update all elements by adding a value
arr = [10, 20, 30, 40, 50]
addValue = 5
for i in range(len(arr)):
    arr[i] += addValue
print("After adding 5 to all elements:", arr)

# Method 5: Update using list comprehension
arr = [10, 20, 30, 40, 50]
arr = [x * 2 for x in arr]
print("After doubling all elements (comprehension):", arr)

# Method 6: Update specific range
arr = [10, 20, 30, 40, 50]
arr[1:4] = [99, 99, 99]
print("After updating range [1:4] to 99:", arr)

# Method 7: Using function to update
def update_element(lst, index, value):
    lst[index] = value
    return lst

arr = [10, 20, 30, 40, 50]
update_element(arr, 2, 100)
print("After using function to update index 2:", arr)
